proposals/p2759.md

```diff
@@ -0,0 +1,162 @@
+suitability specifically with respect to fitness as a C++ successor language.
+
+The features proposed for 0.1 language in turn focus on C++ interoperability and
+a minimal subset of foundational aspects of the language.
+
+Beyond the language itself, the other project features and milestones proposed
+focus on enabling evaluation of the language design and interoperating with C++
+in practice.
+
+## Problem
+
+The initial set of concrete goals for the Carbon project have focused on
+establishing the project itself (governance, evolution, community,
+infrastructure, etc.) and exploratory work on both the language design and
+implementation.
+
+The project is now reasonably healthy and executing effectively in the open, and
+we've made enough exploratory progress on both design and implementation that we
+need to pick a much more specific target to prioritize further work effectively.
+
+The obvious next goal, as outlined in our 2023 roadmap, is to prepare Carbon for
+detailed and in-depth evaluation by potential users to see if it could
+potentially meet their needs for a C++ successor language. While some aspects of
+our potential value proposition can be considered abstractly, many of the hard
+questions around "will it work with my code?" and "how well will it work?" need
+direct evaluation to gain confidence.
+
+We need to prioritize a specific set of features and milestones that are
+sufficient for this evaluation. Once this 0.1 language is "done" we can start
+advocating for in-depth evaluation.
+
+## Background
+
+-   [Carbon's 2023 roadmap](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/roadmap.md)
+
+## Proposal
+
+We suggest a set of initial milestones for the Carbon project with version
+numbers to make them easy to talk about. For each milestone, we try to outline
+what the theme is, provide some concrete goals that should be addressed, and a
+list of concrete features we expect to be complete.
+
+While the proposal only suggests three milestones initially, it isn't precluding
+adding additional milestones when and where they make sense as the project
+progresses.
+
+This proposal is suggesting a concrete and near-complete set of features for the
+first milestone. While this list may be updated based on new information, we are
+reasonably confident in its completeness. However, subsequent milestones are
+intentionally left incomplete and open-ended. They will be filled in as they get
+closer. The intent in having them here is to allow us to explicitly defer things
+to later milestones without leaving them out entirely in cases where there is
+specific interest in knowing when a major feature is expected.
+
+**Summary of proposed milestones:**
+
+-   "0.1" -- the MVP (Minimum Viable Product) language for the _evaluation_ of
+    Carbon by C++ community and developers.
+-   "0.2" -- feature complete for evaluating and concluding the experiment
```

Comment:
```suggestion
-   "0.2" -- feature complete for in-production evaluation and concluding the experiment
```

Reply:
Is there another way to phrase this? I think I largely know what you mean here, but I'm worried about it being confused with "production-ready" below.

Reply:
How about "feature complete for in-production experiments"?

Reply:
That also has a collision ("the experiment" later in this sentence).

Maybe I don't understand what you're trying to get to here after all -- what's the gap that you'd like to close?

Maybe it's more trying to make this clear that this covers the *whole* evaluation space? What about something like:

```suggestion
-   "0.2" -- feature complete for initial users to finish their evaluation and the project to conclude its experiment
```

The thing I'm getting to is that I think whether the evaluation is "in-production" or not is somewhat hard to define and likely to differ for different users. But we can maybe more concretely describe the goal post -- everyone needs to be able to complete their evaluation and make a reasonable confidence decision.

Reply:
Your latest suggestion works for me. 0.1 there's enough there to begin a meaningful evaluation. 0.2 there's enough there to complete a meaningful evaluation.